backbone/src/main/java/org/researchstack/backbone/step/active/recorder/DeviceMotionRecorder.java

```diff
@@ -101,50 +119,167 @@ protected void writeJsonData() {
         if (hasAvailableType(availableSensorList, Sensor.TYPE_ROTATION_VECTOR)) {
             sensorTypeList.add(Sensor.TYPE_ROTATION_VECTOR);
         }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            if (hasAvailableType(availableSensorList, Sensor.TYPE_GAME_ROTATION_VECTOR)) {
+                sensorTypeList.add(Sensor.TYPE_GAME_ROTATION_VECTOR);
+            }
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            if (hasAvailableType(availableSensorList, Sensor.TYPE_GEOMAGNETIC_ROTATION_VECTOR)) {
+                sensorTypeList.add(Sensor.TYPE_GEOMAGNETIC_ROTATION_VECTOR);
+            }
+        }
 
         return sensorTypeList;
     }
 
+
     @Override
-    public void onSensorChanged(SensorEvent sensorEvent) {
-        switch (sensorEvent.sensor.getType()) {
+    public void recordSensorEvent(SensorEvent sensorEvent, JsonObject jsonObject) {
+        int sensorType = sensorEvent.sensor.getType();
+        String sensorTypeKey = SENSOR_TYPE_TO_DATA_TYPE.get(sensorType);
+
+        if (Strings.isNullOrEmpty(sensorTypeKey)) {
+            logger.warn("Unable find type key for sensor type: "
+                    + sensorType);
+            return;
+        }
+
+        jsonObject.addProperty(SENSOR_DATA_TYPE_KEY, sensorTypeKey);
+        jsonObject.addProperty(SENSOR_EVENT_ACCURACY_KEY, sensorEvent.accuracy);
```

Comment:
Shouldn't you add "uptime" here?

Reply:
I put that in SensorRecorder, so everything that extends it will use the same timestamp and uptime.